File: views/src/api/es-index.js
import request from '@/utils/request'

const api = '/api/es_index/'

export function CreateAction(data) {
  return request({
    url: api + 'CreateAction',
    method: 'post',
    data
  })
}

export function GetSettingsAction(data) {
  return request({
    url: api + 'GetSettingsAction',
    method: 'post',
    data
  })
}

export function IndexNamesAction(data) {
  return request({
    url: api + 'IndexNamesAction',
    method: 'post',
    data
  })
}

export function ReindexAction(data) {
  return request({
    url: api + 'ReindexAction',
    method: 'post',
    data
  })
}

export function GetAliasAction(data) {
  return request({
    url: api + 'GetAliasAction',
    method: 'post',
    data
  })
}

File: router/es_index.go
package router

import (
	. "ElasticView/controller"

	"github.com/gin-gonic/gin"
)

// ES索引 路由
func runEsIndex(app *gin.Engine) {
	esIndex := app.Group("/api/es_index")
	{
		esIndex.POST("DeleteAction", EsIndexController{}.DeleteAction)
		esIndex.POST("CreateAction", EsIndexController{}.CreateAction)
		esIndex.POST("GetSettingsAction", EsIndexController{}.GetSettingsAction)
		esIndex.POST("IndexNamesAction", EsIndexController{}.IndexNamesAction)
		esIndex.POST("ReindexAction", EsIndexController{}.ReindexAction)
		esIndex.POST("GetAliasAction", EsIndexController{}.GetAliasAction)
		esIndex.POST("OperateAliasAction", EsIndexController{}.OperateAliasAction)
		esIndex.POST("GetSettingsInfoAction", EsIndexController{}.GetSettingsInfoAction)
		esIndex.POST("StatsAction", EsIndexController{}.StatsAction)

	}
}
